sstables/storage.cc

```diff
@@ -37,6 +37,7 @@ namespace sstables {
 class filesystem_storage final : public sstables::storage {
     sstring dir;
     std::optional<sstring> temp_dir; // Valid while the sstable is being created, until sealed
+    sstables_manager& manager;
```

Comment:
This reference is not needed. E.g. the `s3_storage::delete_with_system_keyspace()` also needs (well, _will_ need for #13567) sstables manager and it's going to get it like `s3_storage::wipe()` does: via `auto& sys_ks = sst.manager().system_keyspace();`

Another (better) option is to just pass the `sstables_manager&` as an argument to `atomic_deleter()` all the more so the only caller is sstables manager itself and it will just pass `*this`